src/Controls/src/Core/Menu/MenuItem.cs

```diff
@@ -31,7 +31,13 @@ public partial class MenuItem : BaseMenuItem, IMenuItemController, IStyleSelecta
 		public static readonly BindableProperty IsDestructiveProperty = BindableProperty.Create(nameof(IsDestructive), typeof(bool), typeof(MenuItem), false);
 
 		/// <summary>Bindable property for <see cref="IconImageSource"/>.</summary>
-		public static readonly BindableProperty IconImageSourceProperty = BindableProperty.Create(nameof(IconImageSource), typeof(ImageSource), typeof(MenuItem), default(ImageSource));
+		public static readonly BindableProperty IconImageSourceProperty = BindableProperty.Create(nameof(IconImageSource), typeof(ImageSource), typeof(MenuItem), default(ImageSource),
+		 propertyChanged: (bindable, oldValue, newValue) => {
```

Comment:
we need to unparent as well
```suggestion
		 propertyChanged: (bindable, oldValue, newValue) => {
		 	if (oldValue is ImageSource oldIcon)
		 		oldIcon.PArent = null;
```

Reply:
Done